docs/reference/commands/run.rst

```diff
@@ -67,19 +67,51 @@ name specified should be the machine-readable package name for the app.
 Update the application's source code before running. Equivalent to running::
 
     $ briefcase update
+    $ briefcase build
+    $ briefcase run
+
+``-r`` / ``--update-requirements``
+----------------------------------
+
+Update application requirements before running. Equivalent to running::
+
+    $ briefcase update -r
+    $ briefcase build
+    $ briefcase run
```

Comment:
It isn't entirely equivalent, because `build -r` and `run -r` will update the requirements without updating the sources. This is a combination that `update` itself doesn't support, and I don't see much use in adding it.

So how about this:
* The update phase would always update the sources, regardless of how it was triggered: `-u`, `-r` or `--update-resources`. 
* `--no-update` would prevent the update phase entirely, with no need to pass `--no-update-requirements` and `--no-update-resources` as well. If `--no-update` was used along with `-r` or `--update-resources`, we would give an error message.

That way, the update phase really would be equivalent to a standalone `update` command, and it would be easier for the user to understand and control.

Reply:
The equivalence issue in documentation is a good point - that's a copy-and-paste that I clearly didn't fully think through.

As for `--no-update`; I went back and forth on whether `--no-update` should be a "disable all updates" or a "disable only code updates". I ended up landing on the "full parity between update and no-update pairs" because I figured these were all power options, so flexibility would be useful; however, on reflection I think you're probably right that it can be simplified (plus, it addresses the "long options" problem I mentioned in the other comment).

To confirm we're on the same page, we're talking about the following behavior:
1. `update` - updates code, but nothing else
2. `update -r` updates code and requirements
3. `update --update-resources` updates code and resources, but not requirements
4. `build` - does no updates
5. `build -u` updates code before building
6. `build -r` updates requirements before building (but *not* code)
7.  `build --test` updates code into test mode before building; no requirements/resources changes
8.  `build --test -r` updates code and requirements into test mode before building; no resources changes
9. `build --test --no-update` updates nothing; just does the build
10. `build --test -r --no-update ` is an error.

4-10 would then have the same behaviour for `run`.